### PR TITLE
Adding additional info in csv

### PR DESCRIPTION
**Pull request by Garima-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 1]
```diff
@@ -266,6 +266,8 @@ class Organization: ADOSVTBase
                                         $controlResult.AddMessage("List of non-ALT accounts: ", $($stateData | Format-Table -AutoSize | Out-String));
                                         $controlResult.SetStateData("List of non-ALT accounts: ", $stateData);
                                         $controlResult.AdditionalInfo += "Count of non-ALT accounts with admin privileges: " + $nonSCCount;
+                                        $formatedSCMembers = $nonSCMembers | ForEach-Object { $_.name + ': '+ $_.mailAddress + ': ' + $_.groupName }
+                                        $controlResult.AdditionalInfoInCSV = "totalAdmin: $($totalAdminCount); totalNonALTAdmin: $($nonSCCount); nonAltAdminList: $(($formatedSCMembers | Select -First 10) -join '; ')"
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:270]: Camel case

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 2]
```diff
@@ -622,22 +632,41 @@ class Organization: ADOSVTBase
...
-                        $inactiveGuestUsers= $inactiveGuestUsers | Select-Object @{Name="DisplayName"; Expression = {$_.DisplayName}},@{Name="MailAddress"; Expression = {$_.MailAddress}}, @{Name="InactiveFromDays"; Expression = {$_.InactiveFromDays}}, @{Name="ProjectReference"; Expression = {$_.ProjectEntitlements.projectref.name}}, @{Name="ProjectPermission"; Expression = {$_.ProjectEntitlements.group.displayName}}, @{Name="AccessLevel"; Expression = {$_.AccessLevel}}
-                        $display = $inactiveGuestUsers |FT -AutoSize | Out-String -Width 512 
-                    }
-                    else {
-                        $display = $inactiveGuestUsers |FT DisplayName,MailAddress,InactiveFromDays -AutoSize | Out-String -Width 512
-                    }
-                    $controlResult.AddMessage($display)
-                }
+                #$inactiveGuestUsers = @($guestListDetailed | Where-Object { $_.InactiveFromDays -eq "User was never active." })
+                #$inactiveCount = $inactiveGuestUsers.Count
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:636]: remove commented part

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 3]
```diff
@@ -655,6 +684,8 @@ class Organization: ADOSVTBase
                         $display = ($activeGuestUsers |FT DisplayName,MailAddress,InactiveFromDays -AutoSize | Out-String -Width 512)
                     }
                     $controlResult.AddMessage($display)
+                    $formatedGuestUsers = ($activeGuestUsers | Sort-Object -Property InactiveFromDays -Descending) | ForEach-Object { $_.DisplayName + ': ' +$_.MailAddress +': '+ $_.InactiveFromDays }
+                    $controlResult.AdditionalInfoInCSV = "totalGuestUsers: $($activeCount); usersList: " + (($formatedGuestUsers | Select -First 10) -join '; ' )
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:688]: Camel case

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 4]
```diff
@@ -802,12 +833,14 @@ class Organization: ADOSVTBase
...
                 if ($disconnectedUsersCount -gt 0 )
                 {
+                    $disconnectedUsersList = @();
                     $disconnectedUsersList += @($responseObj[0].users | Select-Object -Property @{Name = "Name"; Expression = { $_.displayName } }, @{Name = "MailAddress"; Expression = { $_.preferredEmailAddress } })
                     $controlResult.AddMessage("Count of disconnected users: $($disconnectedUsersCount)`n");
                     $controlResult.AddMessage([VerificationResult]::Failed, "Remove access for below disconnected users: ", ($disconnectedUsersList | FT | out-string));
                     $controlResult.SetStateData("Disconnected users list: ", $disconnectedUsersList);
                     $controlResult.AdditionalInfo += "Count of disconnected users: " + $disconnectedUsersCount ;
                     $controlResult.AdditionalInfo += "List of disconnected users: " + $disconnectedUsersList;
+                    $controlResult.AdditionalInfoInCSV = "totalDisconnectedUsers: " + $disconnectedUsersCount;
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:843]: same

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 5]
```diff
@@ -1306,6 +1341,10 @@ class Organization: ADOSVTBase
                     $controlResult.AddMessage("Count of auto-injected extensions: " + $extCount);
                     $controlResult.AdditionalInfo += "Count of auto-injected extensions: " + $extCount;
                     $this.ExtensionControlHelper($controlResult, $autoInjExt, 'AutoInjected')
+                   
+                    $controlResult.AdditionalInfoInCSV += "Count of Auto-Injected Extensions: $($extCount) ; ";
+                    $ExtList = $unknown | ForEach-Object { $_.extensionName + ' by ' + $_.publisherName } | select-object -Unique -First 10;
+                    $controlResult.AdditionalInfoInCSV += "Unknown_Publisher: $($ExtList -join ' ; ');";
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:1347]: remove _

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 6]
```diff
@@ -1353,6 +1392,8 @@ class Organization: ADOSVTBase
                 else{
                     $controlResult.AddMessage([VerificationResult]::Passed,"Number of human administrators configured meet the minimum required administrators count: $($this.ControlSettings.Organization.MinPCAMembersPermissible)");
                 }
+                $controlResult.AdditionalInfoInCSV += "TotalPCA_Admins: $($TotalPCAMembers) ; ";
+                $controlResult.AdditionalInfoInCSV += "MinPCA_Required: $($this.ControlSettings.Organization.MinPCAMembersPermissible) ; ";
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:1396]: same

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 7]
```diff
@@ -1602,35 +1648,30 @@ class Organization: ADOSVTBase
...
                     $inactiveUsersWithDaysCount = ($inactiveUsersWithDays | Measure-Object).Count
                     if($inactiveUsersWithDaysCount -gt 0) {
                         $controlResult.AddMessage("`nCount of guest users who are inactive from last $($GuestUserInactivePeriodInDays) days: $($inactiveUsersWithDaysCount)");
                         $inactiveUsersTable = ($inactiveUsersWithDays | FT | Out-String)
                         $controlResult.AddMessage("Inactive guest users list: `n$inactiveUsersTable");
                         $controlResult.AdditionalInfo += "Count of guest users who are inactive from last $($GuestUserInactivePeriodInDays) days: " + $inactiveUsersWithDaysCount;
                     }
+
+                    $domainsInfo = $inactiveguestUsers | ForEach-Object { $_.user.mailaddress.Split("@")[1]} | select-object -Unique -First 10
+                    $controlResult.AdditionalInfoInCSV  += "Different domains: $($domainsInfo -join ', ')"
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:1664]: Camel case

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1 [hunk 8]
```diff
@@ -1934,6 +1977,8 @@ class Organization: ADOSVTBase
                         $display = ($inactiveUsersWithAdminAccess|FT PrincipalName,DisplayName,Group,LastAccessedDate  -AutoSize | Out-String -Width 512)
                         $controlResult.AddMessage($display)
                         $controlResult.SetStateData("List of inactive users: ", $inactiveUsersWithAdminAccess);
+                        $inactiveUsers = $inactiveUsersWithAdminAccess | ForEach-Object { $_.DisplayName + ': '+$_.PrincipalName+': ' + $_.Group +': ' +$_.LastAccessedDate }
+                        $controlResult.AdditionalInfoInCSV = "totalInactiveAdmin: $($inactiveUsersWithAdminAccess.count); inactiveAdminList: " + (($inactiveUsers | Select -First 10) -join '; ' )
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1:1981]: same

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 [hunk 9]
```diff
@@ -391,13 +391,15 @@ class Project: ADOSVTBase
                 # In case of graph access we will only evaluate the control on the basis of human accounts
                 if($humanAccounts.count -lt $this.ControlSettings.Project.MinPAMembersPermissible){
                     $controlResult.AddMessage([VerificationResult]::Failed,"Number of human administrators configured are less than the minimum required administrators count: $($this.ControlSettings.Project.MinPAMembersPermissible)");
+                    $controlResult.AdditionalInfoInCSV += "Total PA Member Count: $($TotalPAMembers);";
+                    $controlResult.AdditionalInfoInCSV += "Min PA Member required: $($this.ControlSettings.project.MinPAMembersPermissible);";
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:395]: TotalPACount:

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1 [hunk 10]
```diff
@@ -418,6 +420,8 @@ class Project: ADOSVTBase
                 $this.PAMembers = @($this.PAMembers | Select-Object displayName,mailAddress)
                 if($TotalPAMembers -lt $this.ControlSettings.Project.MinPAMembersPermissible){
                     $controlResult.AddMessage([VerificationResult]::Failed,"Number of administrators configured are less than the minimum required administrators count: $($this.ControlSettings.Project.MinPAMembersPermissible)");
+                    $controlResult.AdditionalInfoInCSV += "Total PA Member Count: $($TotalPAMembers);";
+                    $controlResult.AdditionalInfoInCSV += "Min PA Member required: $($this.ControlSettings.project.MinPAMembersPermissible);";
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1:424]: same

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1 [hunk 11]
```diff
@@ -1076,8 +1076,11 @@ class Release: ADOSVTBase
 
                     if(($editableTaskGroups | Measure-Object).Count -gt 0)
                     {
-                        $controlResult.AddMessage("Total number of task groups on which contributors have edit permissions in release definition: ", ($editableTaskGroups | Measure-Object).Count);
-                        $controlResult.AdditionalInfo += "Total number of task groups on which contributors have edit permissions in release definition: " + ($editableTaskGroups | Measure-Object).Count;
+                        $editableTaskGroupsCount = ($editableTaskGroups | Measure-Object).Count;
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Release.ps1:1079]: move outside if

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1 [hunk 12]
```diff
@@ -904,6 +904,9 @@ class Build: ADOSVTBase
                 $controlResult.AddMessage($display)
                 $controlResult.SetStateData("Pipeline code is built from external repository: ",$sourceObj)
            }
+        $controlResult.AdditionalInfo = "$($sourceObj.repositoryName) : $($this.BuildObj[0].repository.type)";
+        $controlResult.AdditionalInfoInCSV = "$($sourceObj.repositoryName) : $($this.BuildObj[0].repository.type)"
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1:908]: Why two AdditionalInfo,AdditionalInfoInCSV fields?

**Review by SaikumarRagala — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**